judge/forms.py

```diff
@@ -20,6 +21,22 @@
 from judge.widgets import HeavyPreviewPageDownWidget, MathJaxPagedownWidget, PagedownWidget, Select2MultipleWidget, \
     Select2Widget
 
+totp_or_scratch_code_validators = [
+    RegexValidator('^[0-9]{6}$', _('Two-factor authentication tokens must be 6 decimal digits.')),
+    RegexValidator('^[A-Z2-7]{16}$', _('Scratch codes must be 16 base32 characters.')),
+]
+
+
+def totp_or_scratch_code_validator(totp_or_scratch_code):
+    err = None
+    for validator in totp_or_scratch_code_validators:
+        try:
+            validator(totp_or_scratch_code)
+            return totp_or_scratch_code
+        except ValidationError as error:
+            err = error
```

Comment:
Won't this basically always return the `'Scratch codes must be 16 base32 characters.'` error? Any input I can think of that fails the 2fa validator will also fail the scratch code validator, unless it's an actually valid scratch code.